src/frontends/pytorch/src/transforms/index_loop_getitem_replacer.cpp

```diff
@@ -0,0 +1,154 @@
+        if (param_targets.size() != 1)
+            return false;
+
+        auto getitem = param_targets.begin()->get_node()->shared_from_this();
+        if (!ov::as_type_ptr<v8::Gather>(getitem))
+            return false;
+
+        auto dim = chunk_op->input_value(2);
+        if (!ov::as_type_ptr<v0::Constant>(dim.get_node_shared_ptr()))
+            return false;
+
+        // connect chunk input directly to loop
+        auto chunk_input = chunk_op->input_value(0);
+        chunk_op->output(0).replace(chunk_input);
+        // len(chunks) is number of iterations
+        auto chunks_outside = chunk_op->input_value(1);
+        loop_op->input_value(0).replace(chunks_outside);
+
+        auto chunk_counter = getitem->input_value(1);
+
+        auto tensor_0 = v0::Constant::create(element::i32, Shape{1}, {0});
+        auto one_1d = v0::Constant::create(element::i32, Shape{1}, {1});
+
+        auto input_shape = std::make_shared<v3::ShapeOf>(chunk_input, element::i32);
+        auto input_dimension = std::make_shared<v8::Gather>(input_shape, dim, tensor_0);
+        auto init_chunk_size = std::make_shared<v1::Divide>(input_dimension, chunks_outside, true);
+
+        // Add 1 if input is not evenly divisible by chunks
+        auto last_chunk_size = std::make_shared<v1::Mod>(input_dimension, chunks_outside);
+        auto is_last_nonzero = std::make_shared<v1::Greater>(last_chunk_size, tensor_0);
+        auto is_last_nonzero_int = std::make_shared<v0::Convert>(is_last_nonzero, element::i32);
+        auto chunk_size = std::make_shared<v1::Add>(init_chunk_size, is_last_nonzero_int);
+        auto dim_1d = std::make_shared<v1::Reshape>(dim, one_1d, false);
+
+        // Add new inputs in Loop: chunk_size and dim_1d
+        auto inp_descs = loop_op->get_input_descriptions();
+        auto chunks_size_body = std::make_shared<v0::Parameter>(element::i32, Shape{1});
+        auto dim_body = std::make_shared<v0::Parameter>(dim.get_element_type(), Shape{1});
+        body->add_parameters({chunks_size_body, dim_body});
+        loop_op->set_argument(loop_op->get_input_size(), chunk_size);
+        loop_op->set_argument(loop_op->get_input_size(), dim_1d);
+        inp_descs.push_back(std::make_shared<ov::op::util::MultiSubGraphOp::InvariantInputDescription>(
+            loop_op->get_input_size() - 2,
+            body->get_parameters().size() - 2));
+        inp_descs.push_back(std::make_shared<ov::op::util::MultiSubGraphOp::InvariantInputDescription>(
+            loop_op->get_input_size() - 1,
+            body->get_parameters().size() - 1));
+        loop_op->set_input_descriptions(0, inp_descs);
+
+        auto start = std::make_shared<v1::Multiply>(chunk_counter, chunks_size_body);
+        auto stop = std::make_shared<v1::Add>(start, chunks_size_body);
+        auto curr_chunk = std::make_shared<v8::Slice>(chunk_param, start, stop, one_1d, dim_body);
+        replace_node(getitem, curr_chunk);
+        copy_runtime_info({chunk_op, getitem},
```

Comment:
I would recommend to use `pass::NodeRegistry` and simplify the transformation a bit:
`pass::NodeRegistry rg;
 // use rg.make<...>() instead of std::make_shared<...>() e.g.
auto start = rg.make<v1::Multiply>(...);
copy_runtime_info({chunk_op, getitem}, rg.get());